### PR TITLE
fix: implement authorization on origDatablock

### PR DESCRIPTION
**Pull request by Junjiequan**
## Description

A user can create an orig datablock only if he/she is allowed to update the dataset specified.

## Tests included/Docs Updated?

- [ ] Included for each change/fix?
- [ ] Passing? (Merge will not be approved unless this is checked) 
- [ ] Docs updated?
- [ ] New packages used/requires npm install? 
- [ ] Toggle added for new features?

### CODE REVIEW

File: src/origdatablocks/origdatablocks.controller.ts
Comment (nitrosx) [inline src/origdatablocks/origdatablocks.controller.ts:1]: Do we have the same tests on the other endpoints?
Also are we testing for all the different use cases?

Comment (Junjiequan) [reply src/origdatablocks/origdatablocks.controller.ts:1]: There's another ticket for creating test cases, I haven't start to working on the ticket yet https://jira.esss.lu.se/browse/SWAP-3334

**Comment by nitrosx** [comment]
@Junjiequan unfortunately I need to review completely the datasets authorization model, than I can complete this PR review.
I think there are inconsistencies between Datasets and OrigDatablocks permissions that needs to be addressed.

File: src/origdatablocks/origdatablocks.controller.ts [hunk 1]
```diff
@@ -263,12 +347,26 @@ export class OrigDatablocksController {
   )
   @Get("/fullfacet")
   async fullfacet(
+    @Req() request: Request,
     @Query() filters: { fields?: string; facets?: string },
   ): Promise<Record<string, unknown>[]> {
+    const user: JWTUser = request.user as JWTUser;
+    const fields: IOrigDatablockFields = JSON.parse(filters.fields ?? "{}");
```
Comment (bpedersen2) [inline src/origdatablocks/origdatablocks.controller.ts:354]: there are a lot of similar blocks like this,
I guess we should try to refactor it to use a common function. 
something like (pseudo-code):
```
function accessGroupsFilterFields( fields_in, user, OrigDataBlock) {
  if (user) {
   ability = ...
  canViewAll = ...
 if ( !canViewAll ){
   ...
     fields_in.userGroups.push(...)
   }
return fields_in
}
```

**Review by bpedersen2 — CHANGES_REQUESTED** [review]
Generally looks fine, but a bit too much code seems duplicated which makes later maintenance harder.

File: src/origdatablocks/origdatablocks.controller.ts (hunk 1 above)
Comment (nitrosx) [reply src/origdatablocks/origdatablocks.controller.ts:354]: @bpedersen2 I will postpone code refactoring to a new issue.
I'm going through a review of all the authorization for datasets and connected entities, which will require some changes.
We chat tomorrow during the meeting if you are present.

Comment (Junjiequan) [reply src/origdatablocks/origdatablocks.controller.ts:354]: I've created new issue to address this issue #821

**Review by nitrosx — APPROVED** [review]

**Pull request merged**